blockprotocol/blockprotocol
# Add `/settings/billing` page

## 🌟 What is the purpose of this PR?




This PR adds a "Billing" settings panel to the user's dashboard.

## 🔗 Related links





- [Asana task](https://app.asana.com/0/1203543021352041/1203756054892119/f) _(internal)_


## 🔍 What does this change?




See commits.

## 📜 Does this require a change to the docs?



Not that I'm aware of.

## 🐾 Next steps



- [add `/upgrade` page to enable users to upgrade their subscription tier](https://app.asana.com/0/1203543021352032/1203655456562773/f)
- [implement "change payment" settings panel](https://app.asana.com/0/0/1203781148500075/f)
- [implement "usage limits" input](https://app.asana.com/0/0/1203781148500077/f)
- [implement "payment history" section of billing panel](https://app.asana.com/0/1203543021352041/1203781148500078/f)
- [refactor `/settings/api-keys` page to be a settings panel](https://app.asana.com/0/0/1203781148500080/f)

## 🛡 What tests cover this?



Manual testing.

## ❓ How to test this?



1. Checkout the branch
2. Add the `NEXT_PUBLIC_BILLING_FEATURE_FLAG=1` env var to your `app/site/.env.local` to enable the billing feature flag
3. Login as a user and navigate to `/settings/billing`
4. You should see the billing page as a user with a "free" subscription tier. The easiest way of seeing the page in a "hobby" or "pro" state is to change the return value of [this react hook](https://github.com/blockprotocol/blockprotocol/blob/246d8197a73da353734bb5f56de2790ebf8ff962/apps/site/src/pages/settings/billing-settings-panel/billing-settings-panel.tsx#L38-L44) locally (either to `"hobby"` or `"pro"`).

## 📹 Demo

<details>
<summary>"free" user billing page</summary>

![localhost_3000_settings_billing (4)](https://user-images.githubusercontent.com/42802102/213779415-797c80a7-a916-4947-b77b-3f5ecf32845f.png)

</details>

<details>
<summary>"hobby" user billing page</summary>

![localhost_3000_settings_billing](https://user-images.githubusercontent.com/42802102/213778764-98c97d58-a5f2-4af8-9a7f-9ec4ce4efb87.png)

</details>

<details>
<summary>"pro" user billing page</summary>

![localhost_3000_settings_billing (3)](https://user-images.githubusercontent.com/42802102/213778970-2171e0b1-6665-4df2-a946-b6a9e920df01.png)

</details>
